psydac/feec/global_projectors.py

```diff
@@ -791,3 +878,32 @@ def evaluate_dofs_3d_3form(
                             F[i1, i2, i3] += \
                                     quad_w1[i1, g1] * quad_w2[i2, g2] * quad_w3[i3, g3] * \
                                     f(quad_x1[i1, g1], quad_x2[i2, g2], quad_x3[i3, g3])
+
+#------------------------------------------------------------------------------
+def evaluate_dofs_3d_vec(
+        intp_x1, intp_x2, intp_x3, # interpolation points
+        F1, F2, F3,                # array of degrees of freedom (intent out)
+        f1, f2, f3,                # input scalar function (callable)
+        ):
+    
+    # evaluate input functions at interpolation points (make sure that points are in [0, 1])
+    
+    
```

Comment:
Usually two consecutive empty lines are only used between separate functions or classes. These can be removed here completely.